assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'objParam is required.' );
+
+				dispatch.fetchGetSomeData( {} );
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'aParam is required.' );
+
+				consoleErrorSpy.mockClear();
+			} );
+
+			it( 'yields the expected actions for an arguments error', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+					argsToParams: ( requiredParam ) => {
+						invariant( requiredParam, 'requiredParam is required.' );
+						return {
+							requiredParam,
+						};
+					},
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				// Catch invariant to get exactly the error we expect.
+				let error;
+				try {
+					invariant( false, 'requiredParam is required.' );
+				} catch ( err ) {
+					error = err;
+				}
+
+				expect( action.next().value ).toEqual( {
+					response: undefined,
+					error,
+				} );
+			} );
+
+			it( 'yields the expected actions for a success request', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.next( 42 ).value.type ).toEqual( 'RECEIVE_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FINISH_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value ).toEqual( {
+					response: 42,
+					error: undefined,
+				} );
+			} );
+
+			it( 'yields the expected actions for an error request', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				const error = {
+					code: 'this-went-wrong',
+					message: 'This went wrong.',
+				};
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.throw( error ).value.type ).toEqual( 'CATCH_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value ).toEqual( {
+					response: undefined,
+					error,
+				} );
+			} );
+
+			it( 'makes a network request based on controlCallback', async () => {
+				const expectedResponse = 'response-value';
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( expectedResponse ),
+						{ status: 200 }
+					);
+
+				const { response, error } = await dispatch.fetchGetSomeData( {}, 'value-to-key-response-by' );
```

Comment:
This threw me off for a bit. It's not immediately obvious that the second parameter becomes `aParam` which is used in the reducer. Maybe this is just because the store definition is so far away from this spot, but just thought I'd  mention it.

Reply:
Yeah it's probably because the definition is so far away. What happens with these arguments is entirely up to the `argsToParams` logic.